src/interceptors/fetch/index.ts

```diff
@@ -62,10 +71,7 @@ export class FetchInterceptor extends Interceptor<HttpRequestEventMap> {
       })
 
       if (resolverResult.error) {
-        const error = Object.assign(new TypeError('Failed to fetch'), {
-          cause: resolverResult.error,
-        })
-        return Promise.reject(error)
+        return Promise.reject(resolverResult.error)
```

Comment:
I believe this is incorrect. I copied the previous behavior from Undici and we must comply by it. Note that the FetchInterceptor is primarily meant for Node, and I trust Undici implement the spec rather faithfully.

Reply:
I'd suggest we revert this particular change for now because it's not related to the abort controller support. We can discuss it as a separate improvement point, what do you think?

Reply:
I checked the Undici implementation and there are only two causes of error : abortion and network issue.
Since the abortion error is well defined (specific class and specific error code), we can check the error type and reject accordingly. 
If we don't do that, we deviate from production behavior.

Reply:
I agree, we should do it the same way: handle the two error scenarios separately:

- Keep what you've introduced for abort errors.
- Revert what was there previously (the `TypeError`) to handle all the other errors (effectively, network errors).

Reply:
Thanks so much for addressing it so quickly! Will give it the last round of review and let's get this published.